internal/keppel/manifest.go

```diff
@@ -24,158 +24,235 @@ import (
+	return append(references, a.m.LayerInfos()...)
 }
 
-func (a ociManifestAdapter) ManifestReferences(pf models.PlatformFilter) []manifestlist.ManifestDescriptor {
+func (a v2ManifestAdapter) ManifestReferences(pf models.PlatformFilter) []imageSpecs.Descriptor {
 	return nil
 }
 
-func (a ociManifestAdapter) AcceptableAlternates(pf models.PlatformFilter) []manifestlist.ManifestDescriptor {
+func (a v2ManifestAdapter) AcceptableAlternates(pf models.PlatformFilter) []imageSpecs.Descriptor {
 	return nil
 }
 
-// listManifestAdapter provides the ParsedManifest interface for the contained type.
-type listManifestAdapter struct {
-	m *manifestlist.DeserializedManifestList
+// v2ManifestListAdapter provides the ParsedManifest interface for the contained type.
+type ociIndexAdapter struct {
+	m *manifest.OCI1Index
 }
 
-func (a listManifestAdapter) FindImageConfigBlob() *distribution.Descriptor {
+func (a ociIndexAdapter) FindImageConfigBlob() *types.BlobInfo {
 	return nil
 }
 
-func (a listManifestAdapter) FindImageLayerBlobs() []distribution.Descriptor {
+func (a ociIndexAdapter) FindImageLayerBlobs() []manifest.LayerInfo {
 	return nil
 }
 
-func (a listManifestAdapter) BlobReferences() []distribution.Descriptor {
+func (a ociIndexAdapter) BlobReferences() []manifest.LayerInfo {
 	return nil
 }
 
-func (a listManifestAdapter) ManifestReferences(pf models.PlatformFilter) []manifestlist.ManifestDescriptor {
-	result := make([]manifestlist.ManifestDescriptor, 0, len(a.m.Manifests))
+func (a ociIndexAdapter) ManifestReferences(pf models.PlatformFilter) []imageSpecs.Descriptor {
+	result := make([]imageSpecs.Descriptor, 0, len(a.m.Manifests))
 	for _, m := range a.m.Manifests {
-		if pf.Includes(m.Platform) {
+		if pf.Includes(*m.Platform) {
 			result = append(result, m)
 		}
 	}
 	return result
 }
 
-func (a listManifestAdapter) AcceptableAlternates(pf models.PlatformFilter) []manifestlist.ManifestDescriptor {
-	var result []manifestlist.ManifestDescriptor
+func (a ociIndexAdapter) AcceptableAlternates(pf models.PlatformFilter) []imageSpecs.Descriptor {
+	var result []imageSpecs.Descriptor
 	for _, m := range a.ManifestReferences(pf) {
 		// If we have an application/vnd.docker.distribution.manifest.list.v2+json manifest, but the
 		// client only accepts application/vnd.docker.distribution.manifest.v2+json, in order to stay
 		// compatible with the reference implementation of Docker Hub, we serve this case by recursing
 		// into the image list and returning the linux/amd64 manifest to the client.
 		//
 		// This case is relevant for the support of tagged multi-arch images in `docker pull`.
-		if a.m.Versioned.MediaType == manifestlist.MediaTypeManifestList && m.MediaType == schema2.MediaTypeManifest {
+		if a.m.MediaType == manifest.DockerV2ListMediaType && m.MediaType == manifest.DockerV2Schema2MediaType {
 			if m.Platform.OS == "linux" && m.Platform.Architecture == "amd64" {
 				result = append(result, m)
 			}
 		}
```

Comment:
This `if` branch is unreachable on this type because this is a function on `ociIndexAdapter`.